src/Haskell/Ide/Engine/Plugin/HieExtras.hs

```diff
@@ -141,16 +159,15 @@ instance ModuleCache CachedCompletions where
         showModName :: ModuleName -> T.Text
         showModName = T.pack . moduleNameString
 
-#if __GLASGOW_HASKELL__ >= 802
         asNamespace :: ImportDecl name -> ModuleName
         asNamespace imp = fromMaybe (iDeclToModName imp) (fmap GHC.unLoc $ ideclAs imp)
-#else
-        asNamespace :: ImportDecl name -> ModuleName
-        asNamespace imp = fromMaybe (iDeclToModName imp) (ideclAs imp)
-#endif
+
         -- Full canonical names of imported modules
         importDeclerations = map unLoc limports
 
+        -- The list of all importable Modules from all packages
+        moduleNames = map showModName (GM.listVisibleModuleNames GHC.unsafeGlobalDynFlags)
```

Comment:
Is getting global dynamic flags ok here?

Reply:
Is it possible to use `getDynFlags` in `PluginUtils.hs` instead? https://github.com/haskell/haskell-ide-engine/blob/60d6bd40c295cbd3a7443fa7735d4f47dce8b0cd/src/Haskell/Ide/Engine/Plugin/HieExtras.hs#L55-L58